core/src/locktower.rs

```diff
@@ -162,8 +162,17 @@ impl Locktower {
                 };
                 Self::update_ancestor_lockouts(&mut stake_lockouts, &vote, ancestors);
             }
-            // each account hash a stake for all the forks in the active tree for this bank
-            Self::update_ancestor_stakes(&mut stake_lockouts, bank_slot, lamports, ancestors);
+
+            if vote_state.votes.len() > 1 {
+                let last_real_vote = &vote_state.votes[vote_state.votes.len() - 2];
```

Comment:
That 2 seems arbitrary to me. Are you planning to add a test before this goes in?

Reply:
should be `nth_recent_vote(1)`

Reply:
can you add a comment explaining that `0` is the simulated vote.

Reply:
sure thing